### PR TITLE
Remove CURRENT from the list of released versions

### PR DESCRIPTION
**Pull request by nik9000**
It was only in the list because it was required for `randomVersionBetween` but now (#25042) it isn't. Since `CURRENT` isn't actually released, we should stop confusing ourselves by saying that it is.

### CODE REVIEW

File: test/framework/src/main/java/org/elasticsearch/test/VersionUtils.java [hunk 1]
```diff
@@ -149,9 +147,13 @@
      * Get the released version before {@code version}.
      */
     public static Version getPreviousVersion(Version version) {
-        int index = RELEASED_VERSIONS.indexOf(version);
-        assert index > 0;
-        return RELEASED_VERSIONS.get(index - 1);
+        for (int i = RELEASED_VERSIONS.size() - 1; i >= 0; i--) {
```
Comment (nik9000) [inline test/framework/src/main/java/org/elasticsearch/test/VersionUtils.java]: This change is required for `getPreviousVersion(CURRENT)` to work. We do that fairly frequently.

File: test/framework/src/main/java/org/elasticsearch/test/VersionUtils.java [hunk 2]
```diff
@@ -168,12 +170,13 @@ public static Version getPreviousVersion() {
      * where the minor version is less than the currents minor version.
      */
     public static Version getPreviousMinorVersion() {
-        Version version = Version.CURRENT;
-        do {
-            version = getPreviousVersion(version);
-            assert version.before(Version.CURRENT);
-        } while (version.minor == Version.CURRENT.minor);
-        return version;
+        for (int i = RELEASED_VERSIONS.size() - 1; i >= 0; i--) {
```
Comment (nik9000) [inline test/framework/src/main/java/org/elasticsearch/test/VersionUtils.java]: I did this because I was upset that it was so inefficient.

**Review by rjernst — APPROVED** [review]
LGTM

**Comment by nik9000** [comment]
Thanks for reviewing @rjernst!

**Comment by nik9000** [comment]
Thanks again for reviewing @rjernst! I finally got around to merging.

**Pull request merged**